File: widgets/sprite.lua
local _context = getContext()
local _painter = _context:getPainter()
local _style = _painter:getStyle()

exports('sprite', function(dict, name, w, h)
	_context:beginDraw(w, h)

	_painter:setColor(_style.sprite.color)
	_painter:drawSprite(dict, name, w, h)

	_context:endDraw()
end)

File: src/items/sprite.ts
import { getFrameChecked } from '../core/frame';
import { Style } from '../core/style';

/**
 * @category Items
 */
export function sprite(dict: string, name: string, w: number, h: number, r = 255, g = 255, b = 255, a = 255) {
	const frame = getFrameChecked();

	const painter = frame.getPainter();

	frame.beginItem(w, h);

	painter.setColor([r, g, b, a]);
	painter.drawSprite(dict, name, w, h);

	frame.endItem();
}
